esp/esdllib/esdl_script.cpp

```diff
@@ -0,0 +1,914 @@
+            if (!ignoreCodingErrors)
+                throw;
+        }
+        return match;
+    }
+};
+
+class CEsdlTransformOperationChoose : public CEsdlTransformOperationBase
+{
+private:
+    IArrayOf<IEsdlTransformOperation> m_conditionals;
+
+public:
+    CEsdlTransformOperationChoose(IPropertyTree * tree, const StringBuffer &prefix) : CEsdlTransformOperationBase(tree, prefix)
+    {
+        if (tree)
+        {
+            loadWhens(tree, prefix);
+            loadOtherwise(tree, prefix);
+        }
+    }
+
+    virtual ~CEsdlTransformOperationChoose(){}
+
+    bool process(IEspContext * context, IPropertyTree *request, IXpathContext * xpathContext) override
+    {
+        ForEachItemIn(i, m_conditionals)
+        {
+            if (m_conditionals.item(i).process(context, request, xpathContext))
+                return true;
+        }
+        return false;
+    }
+
+    virtual void toDBGLog () override
+    {
+    #if defined(_DEBUG)
+        DBGLOG (">>>>>>>>>>> %s >>>>>>>>>>", tagname.str());
+        ForEachItemIn(i, m_conditionals)
+            m_conditionals.item(i).toDBGLog();
+        DBGLOG (">>>>>>>>>>> %s >>>>>>>>>>", tagname.str());
+    #endif
+    }
+
+private:
+    void loadWhens(IPropertyTree * tree, const StringBuffer &prefix)
+    {
+        StringBuffer xpath;
+        Owned<IPropertyTreeIterator> children = tree->getElements(makeOperationTagName(xpath, prefix, "when"));
+        ForEach(*children)
+            m_conditionals.append(*new CEsdlTransformOperationConditional(&children->query(), prefix));
+    }
+
+    void loadOtherwise(IPropertyTree * tree, const StringBuffer &prefix)
+    {
+        StringBuffer xpath;
+        IPropertyTree * otherwise = tree->queryPropTree(makeOperationTagName(xpath, prefix, "otherwise"));
+        if (!otherwise)
+            return;
+        m_conditionals.append(*new CEsdlTransformOperationConditional(otherwise, prefix));
+    }
+};
+
+void processServiceAndMethodTransforms(std::initializer_list<IEsdlCustomTransform *> const &transforms, IEspContext * context, IPropertyTree *tgtcfg, const char *service, const char *method, const char* reqtype, StringBuffer & request, IPropertyTree * bindingCfg)
+{
+    LogLevel level = LogMin;
+    if (!transforms.size())
+        return;
+    if (tgtcfg)
+        level = (unsigned) tgtcfg->getPropInt("@traceLevel", level);
+
+    if (request.length()!=0)
+    {
+        if (level >= LogMax)
+        {
+            DBGLOG("ORIGINAL REQUEST: %s", request.str());
+            StringBuffer marshalled;
+            if (bindingCfg)
+                toXML(bindingCfg, marshalled.clear());
+            DBGLOG("BINDING CONFIG: %s", marshalled.str());
+            if (tgtcfg)
+                toXML(tgtcfg, marshalled.clear());
+            DBGLOG("TARGET CONFIG: %s", marshalled.str());
+        }
+
+        bool strictParams = bindingCfg ? bindingCfg->getPropBool("@strictParams", false) : false;
```

Comment:
As this is pulled from the 'static' esp binding configuration, not the ESDL service binding, we should consider how it will  be set as true for the default. Depending on the timing of our related ESP configuration modularization changes, we may want to update configmgr in tandem to do so.

Reply:
I might have put it here instead of the script tag itself because of the way scripts are merged together.  I really want to stop doing that, and just run each one separately.

Reply:
Let's fix this when we solve ordering, since I think we should also stop merging the transforms together then.  Once each transform is independent this should be part of the transform.  Really we can detect its modern and always turn on strict.